packages/compass-context-menu/src/context-menu-provider.tsx

```diff
@@ -100,7 +92,8 @@ export function ContextMenuProvider({
         capture: true,
       });
     };
-  }, [disabled, handleClosingEvent, handleContextMenuOpen, parentContext]);
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [disabled, handleClosingEvent, parentContext]);
```

Comment:
@gribnoysup is removing it from the dependencies a good enough solution? I was also thinking of wrapping it in a `useCallback` with `[]` but that doesn't seem any better. alternatively I could do a `useCallback on application-level

Reply:
In other parts of the app we use a `useRef` that updates in render to keep the latest stable reference to the prop ([recent example](https://github.com/mongodb-js/compass/blob/ef92780b95c94c59fd9198da912434f664faa7f6/packages/compass-data-modeling/src/components/drawer/relationship-drawer-content.tsx#L69-L70)). I personally do prefer this to just adding an exception (because an exception doesn't differenciate and if you change dependencies later you might miss actually important dependencies) and it means you are actually calling the latest version of the function, but an exception is also fine in this particular case, I'd just suggest to add a comment explaining why you are adding an exception here.